source/fe/mapping_p_generic.cc

```diff
@@ -3824,275 +2111,20 @@ MappingPGeneric<dim, spacedim>::transform(
-        tmp_points[v] = a[GeometryInfo<3>::face_to_cell_vertices(face_no, v)];
-      if (polynomial_degree > 1)
-        for (unsigned int line = 0; line < GeometryInfo<2>::lines_per_cell;
-             ++line)
-          for (unsigned int i = 0; i < polynomial_degree - 1; ++i)
-            tmp_points[4 + line * (polynomial_degree - 1) + i] =
-              a[GeometryInfo<3>::vertices_per_cell +
-                (polynomial_degree - 1) *
-                  GeometryInfo<3>::face_to_cell_lines(face_no, line) +
-                i];
-
-      const std::size_t n_rows =
-        support_point_weights_perimeter_to_interior[1].size(0);
-      a.resize(a.size() + n_rows);
-      auto a_view = make_array_view(a.end() - n_rows, a.end());
-      face->get_manifold().get_new_points(
-        make_array_view(tmp_points.begin(), tmp_points.end()),
-        support_point_weights_perimeter_to_interior[1],
-        a_view);
-    }
-}
-
-
-
-template <>
-void
-MappingPGeneric<2, 3>::add_quad_support_points(
-  const Triangulation<2, 3>::cell_iterator &cell,
-  std::vector<Point<3>> &                   a) const
-{
-  std::array<Point<3>, GeometryInfo<2>::vertices_per_cell> vertices;
-  for (const unsigned int i : GeometryInfo<2>::vertex_indices())
-    vertices[i] = cell->vertex(i);
-
-  Table<2, double> weights(Utilities::fixed_power<2>(polynomial_degree - 1),
-                           GeometryInfo<2>::vertices_per_cell);
-  for (unsigned int q = 0, q2 = 0; q2 < polynomial_degree - 1; ++q2)
-    for (unsigned int q1 = 0; q1 < polynomial_degree - 1; ++q1, ++q)
-      {
-        Point<2> point(line_support_points.point(q1 + 1)[0],
-                       line_support_points.point(q2 + 1)[0]);
-        for (const unsigned int i : GeometryInfo<2>::vertex_indices())
-          weights(q, i) = GeometryInfo<2>::d_linear_shape_function(point, i);
-      }
-
-  const std::size_t n_rows = weights.size(0);
-  a.resize(a.size() + n_rows);
-  auto a_view = make_array_view(a.end() - n_rows, a.end());
-  cell->get_manifold().get_new_points(
-    make_array_view(vertices.begin(), vertices.end()), weights, a_view);
-}
-
-
-
-template <int dim, int spacedim>
-void
-MappingPGeneric<dim, spacedim>::add_quad_support_points(
-  const typename Triangulation<dim, spacedim>::cell_iterator &,
-  std::vector<Point<spacedim>> &) const
-{
-  Assert(false, ExcInternalError());
-}
-
-
-
 template <int dim, int spacedim>
 std::vector<Point<spacedim>>
 MappingPGeneric<dim, spacedim>::compute_mapping_support_points(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell) const
 {
   // get the vertices first
-  std::vector<Point<spacedim>> a;
-  a.reserve(Utilities::fixed_power<dim>(polynomial_degree + 1));
-  for (const unsigned int i : GeometryInfo<dim>::vertex_indices())
-    a.push_back(cell->vertex(i));
+  std::vector<Point<spacedim>> a(cell->n_vertices());
+
+  for (const unsigned int i : cell->vertex_indices())
+    a[i] = cell->vertex(i);
```

Comment:
How should we extend the method `MappingPGeneric<dim, spacedim>::compute_mapping_support_points()` to higher degrees?

Reply:
I don't see how it is related to #10163. As far as that patch goes - I would prefer to load the data into a finite element field and use that instead of relying on an ad-hoc mapping class. If we got that to work from VTK input then we would automatically support this case too.

Reply:
We should definitely not worry about higher-order yet since there is no way to generalize Gauss-Lobatto to simplices. I suspect the higher-order solution will be hierarchical, not nodal - if I recall correctly nek uses hierarchical finite elements.